### PR TITLE
move to a single package.json setup

### PR DESCRIPTION
**Pull request by HyperSprite**
Based on discussion from #1827

### CODE REVIEW

**Comment by vikr01** [comment]
Can you separate commits 095c95426d30a6274830a2c700165d25fdab9726 and dbd0ac321cfd45a7f8f7996e5be5d028cea7715f into separate pull requests? Would like for this to just be for the single `package.json` setup.

**Comment by HyperSprite** [comment]
Removed excess changes.

**Comment by vikr01** [comment]
Rebased against `master`, old commit: dbd0ac321cfd45a7f8f7996e5be5d028cea7715f.

**Comment by amilajack** [comment]
We need to maek sure this works with native deps, which is one of the main reasons why we're using 2 package structure

File: configs/webpack.config.base.js [hunk 1]
```diff
@@ -62,7 +35,7 @@ export default {
    */
   resolve: {
     extensions: ['.js', '.jsx', '.json'],
-    modules: [path.join(__dirname, '..', 'app'), 'node_modules']
+    modules: [path.join(__dirname, '..'), 'node_modules']
```
[the remark this answers is not in the document]
Comment (HyperSprite) [reply configs/webpack.config.base.js]: I just commented this line entirely and did the following for both this plain repo and my personal project that I migrated over the weekend to the same setup. That project includes sqlite/sequelize/umzug (native dep) and they both complete and run.
```bash
rm -rf node_moduels
yarn
yarn test
yarn package
``` 

I am only testing exe and appimage releases though.

**Comment by HyperSprite** [comment]
@amilajack native deps where why I was looking into single package.json setup in the first place. There's a whole discussion in #1827, TL;DR electron-builder fixed this a little while ago and my local testing with SQLite/Sequelize seems to bear that out.

**Comment by vikr01** [comment]
Cherry-picked dbd0ac321cfd45a7f8f7996e5be5d028cea7715f back on due to CI failing because of linting.

Seems `test-e2e-*` scripts will need to be changed as well.

**Comment by aguynamedben** [comment]
@vikr01 I submitted a PR to your branch that fixes paths for the `yarn test-e2e` command that's causing the build to fail.
https://github.com/HyperSprite/electron-react-boilerplate/pull/1

Thanks for your work on this! I think it's going to make rebuilding SQLite+SQLCipher a lot simpler in my project.

File: configs/webpack.config.base.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (aguynamedben) [reply configs/webpack.config.base.js]: So the conclusion here that this line **can** go away, correct? I'll also test removing this on my project where I'm incorporating these changes.

**Comment by HyperSprite** [comment]
@aguynamedben Can you also look at the [`appveyor.yml`](https://github.com/electron-react-boilerplate/electron-react-boilerplate/blob/master/appveyor.yml#L16) file. It might be why the appveyor test are failing. Sorry, I'm away from my computer.

**Comment by aguynamedben** [comment]
@HyperSprite Yep I think you're correct, removed https://github.com/HyperSprite/electron-react-boilerplate/pull/3

**Comment by vikr01** [comment]
@HyperSprite @aguynamedben Do either of you happen to have a public repo that uses this with a native module?

**Comment by HyperSprite** [comment]
I was going to say no... I was waiting to publish it until this PR was merged with master but what the heck. 

In a nutshell it has the following changes:
- Single package.json setup
- Two separate Webpack bundles
- Working Sequelize Models and Migrations
- On the fly DB file creation
- DB file backup on close
- DB Tests
- Hidden DB renderer window with IPC communication through UI renderer.
- Works in `yarn run dev` and `yarn run package`, not `yarn run start`.

You can read all about it on the readme 
https://github.com/HyperSprite/electron-react-boilerplate/tree/single-package-setup-with-sequelize

**Comment by aguynamedben** [comment]
@HyperSprite This is epic. I've been using this in my project and it works great. Thanks for jumping on this.

One note: The path 'app/node_modules' should be removed from .gitignore ([here](https://github.com/electron-react-boilerplate/electron-react-boilerplate/blob/master/.gitignore#L29)). After I implemented these changes in our project, I manually blew away app/node_modules. But much later, when another engineer at my company pulled the branch with these changes, it took us several hours to notice that his local repo was still importing app/node_modules/sqlite3 instead of node_modules/sqlite3. Removing app/node_modules from .gitignore would've helped us notice sooner because we would've realized via `git status` that app/node_modules no longer belongs there.

I will submit a 1-line PR that does this.

**Comment by vikr01** [comment]
@aguynamedben I think even if you were to remove `app/node_modules` from `.gitignore`, it would still be ignored since `node_modules` in `.gitignore` is directory agnostic (if you want to change this, make it `/node_modules`).

In general it's good practice to clean out `node_modules` when messing with `package.json` files (I usually [clean out all ignored files](https://git-scm.com/docs/git-clean#git-clean--X) unless there's something important in there). Unfortunately, it is something easy to miss.

**Comment by aguynamedben** [comment]
@vikr01 Good catch on that detail. PR is submitted here: https://github.com/electron-react-boilerplate/electron-react-boilerplate/pull/1890/files

No biggie if you don't think this is worth merging, just trying to give back.

**Pull request merged**